core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -49,6 +49,32 @@ import scala.collection.{Seq, mutable}
 class KafkaZkClient(zooKeeperClient: ZooKeeperClient, isSecure: Boolean) extends Logging {
   import KafkaZkClient._
 
+  def createSequentialPersistentPath(path: String, data: String = ""): String = {
+    val createRequest = CreateRequest(path, data.getBytes("UTF-8"), acls(path), CreateMode.PERSISTENT_SEQUENTIAL)
+    val createResponse = retryRequestUntilConnected(createRequest)
+    if (createResponse.resultCode != Code.OK) {
```

Comment:
We don't really need the if statement.

Reply:
I see similar code in createConfigChangeNotification() on line 272.
Can you clarify how the if statement can be skipped ?

Reply:
We just need createResponse.resultException.foreach(e => throw e) since createResponse.resultException is None only when createResponse.resultCode is OK.

Mani will have a separate PR to make existing usage consistent.